tests/pytests/unit/modules/test_ansiblegate.py

```diff
@@ -131,3 +131,75 @@ def test_virtual_function(resolver):
     """
     with patch("salt.modules.ansiblegate.ansible", None):
         assert ansible.__virtual__() == "ansible"
+
+
+def test_ansible_module_call(resolver):
+    """
+    Test Ansible module call from ansible gate module
+    :return:
+    """
+
+    class Module:
+        """
+        An ansible module mock.
+        """
+
+        __name__ = "one.two.three"
+        __file__ = "foofile"
+
+        def main():  # pylint: disable=no-method-argument
+            pass
+
+    ANSIBLE_MODULE_ARGS = '{"ANSIBLE_MODULE_ARGS": ["arg_1", {"kwarg1": "foobar"}]}'
+
+    proc = MagicMock(
+        side_effect=[
+            MockTimedProc(stdout=ANSIBLE_MODULE_ARGS.encode(), stderr=None),
+            MockTimedProc(stdout=b'{"completed": true}', stderr=None),
+        ]
+    )
+
+    with patch.object(ansible, "_resolver", resolver), patch.object(
+        ansible._resolver, "load_module", MagicMock(return_value=Module())
+    ):
+        _ansible_module_caller = ansible.AnsibleModuleCaller(ansible._resolver)
+        with patch("salt.utils.timed_subprocess.TimedProc", proc):
+            ret = _ansible_module_caller.call("one.two.three", "arg_1", kwarg1="foobar")
+            proc.assert_any_call(
+                [sys.executable, "foofile"],
+                stdin=ANSIBLE_MODULE_ARGS,
+                stdout=-1,
+                timeout=1200,
+            )
+            try:
+                proc.assert_any_call(
+                    [
+                        "echo",
+                        '{"ANSIBLE_MODULE_ARGS": {"kwarg1": "foobar", "_raw_params": "arg_1"}}',
+                    ],
+                    stdout=-1,
+                    timeout=1200,
+                )
+            except AssertionError:
+                proc.assert_any_call(
+                    [
+                        "echo",
+                        '{"ANSIBLE_MODULE_ARGS": {"_raw_params": "arg_1", "kwarg1": "foobar"}}',
+                    ],
+                    stdout=-1,
+                    timeout=1200,
+                )
+            assert ret == {"completed": True, "timeout": 1200}
+
+
+@patch("salt.utils.path.which", MagicMock(return_value=True))
```

Comment:
Could you please move this mock object into the function level scope. Having mocks in the module level scope could produce issues.

Reply:
Ok, good to know, thanks! I've pushed the requested changes :+1:

Reply:
Oh wait, I missed few of them. 1 sec.

Reply:
Ok, it should be ready now.